### ISSUE TITLE
direnv use_nix relies on `nix-shell ... direnv dump` stdout

### ISSUE THREAD

**Issue by berdario** [direnv/direnv#225]
But nix-shell might print a lot of other unrelated stuff, breaking `direnv load`

**Comment by zimbatm** [comment]
Hi, I had a look over the weekend but the fix is a bit more complex than I thought. Basically it's quite hard to implement redirection on a 3rd file-descriptor only with bash. I have another idea on how to implement a fix by extending `direnv load` directly which should work ok though.

**Comment by cransom** [comment]
I happened to run into this as well when an rc script also dumped status messages to stdout in nix-shell when in `shellHooks`. I redirected it to stderr (`1>&2`) which was a good enough work around for me.

**Comment by zimbatm** [comment]
What's interesting is that nix-build already redirects everything to stderr. If the nix-shell instructions could be translated to nix-build it would solve that problem as well.

**Comment by doronbehar** [comment]
And perhaps switching to `nix run` would also be more reliable right? As proposed [here](https://github.com/direnv/direnv/issues/474#issuecomment-470958133).

**Comment by zimbatm** [comment]
Yeah but `nix run` doesn't know how to load the `shell.nix` file. We would need to introduce a new convention like `env.nix` that lists all the dependencies as an attribute set.
